design/specs/list-data-type.md

```diff
@@ -0,0 +1,194 @@
+dateCreated: 2022-04-06 18:02:58
+---
+
+# Context
+
+This document specifies the design of List data type within the Mathesar application. List data type can include one or more items that are in the same basic data type (e.g., Number), separated by a comma delimiter if more than one item is present. Mathesar aims to support all basic data types in List format. Users will be able to view data that is currently in Mathesar List data type, edit and filter List data, create new columns in List data type, and convert existing data into List data type. 
+
+The goal of the design in general is to help non-technical understand List data type and maintain List data integrity. Aggregating data into Lists via mapping tables is not beyond the scope of this spec and should be designed in visually different ways to communicate the affordance.
+
+Mathesar List data type is similar to [PostgreSQL "Arrays" concept](https://www.postgresql.org/docs/current/arrays.html). However, Mathesar List data type is only supporting single dimensional Array/List currently.
+
+Read [Design for List data type](https://github.com/centerofci/mathesar/issues/978) for a more in-depth explanation of this feature.
+
+Read [Design for visual query builder](https://github.com/centerofci/mathesar/issues/1065) for mapping table details.
+
+# Scenarios
+
+## 1. Working with Existing List Data
+
+Users will be able to view, filter, and edit existing List data. This scenario assumes no List item data type change (e.g., converting a Number item to a Text item) during the edit. 
+
+### 1.1 Viewing Existing List Data
+
+Existing Mathesar List data distinguishes itself from other data type through four unique ways: 
+
+1. Column header data type icon.
+2. Data type icon and description in the column header menu. `Allow List` option is required to be turned on for List data type.
+3. Tooltip information when hovering on the column header data type icon. 
+4. Items in a single cell is separated by comma delimiters and a `show all` option is present. Users can use `show all` function to expand the List and view all List items.
+
+---
+Wireframes
+
+[Viewing Existing List Data](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.59&scaling=min-zoom&starting-point-node-id=7379%3A82626&show-proto-sidebar=1)
+
+---
+
+### 1.2 Filtering Existing List Data
+
+The following filters are available for List data:
+
+1. is empty
+2. is not empty
+3. contains `<LIST ITEM DATA TYPE>`
+4. number of items greater than `<NUMBER>`
+5. number of items greater than or equal to `<NUMBER>`
+6. number of items equal to `<NUMBER>`
+7. number of items lesser than `<NUMBER>`
+8. number of items lesser than or equal to `<NUMBER>`
+
+Users are able to utilize multiple filters through the `Add Filter` feature. By default, the relationship between filters is set to `difference`. Users can enable the `Union` toggle to include data cells that appear in any of the filters.
```

Comment:
We already have an "and"/"or" option in our current filters, can we use that instead of introducing a new concept for union and difference?

Reply:
Got it. I can remove this description. I don't think these features are shown in the prototype I'm using but I could have missed them.